File: service/db/src/main/java/org/delivery/db/userorder/UserOrderRepository.java
package org.delivery.db.userorder;

import java.util.List;
import org.delivery.db.userorder.enums.UserOrderStatus;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserOrderRepository extends JpaRepository<UserOrderEntity, Long> {

    public List<UserOrderEntity> findAllByUserIdAndStatusOrderByIdDesc(Long userId, UserOrderStatus status);

    List<UserOrderEntity> findAllByUserIdAndStatusInOrderByIdDesc(Long userId, List<UserOrderStatus> statusList);

    UserOrderEntity findByIdAndUserId(Long id, Long userId);
    /**
     * SELECT * FROM user-order WHERE user-id = val.user-id AND status IN ('ACCEPT', 'ORDER', ...)
     */
//    public List<UserOrderEntity> findAllByUserIdAndStatusIn(Long userId, List<UserOrderStatus> userOrderStatuses);
}

File: service-security-kotlin-msa/db/src/main/kotlin/org/delivery/db/userorder/UserOrderRepository.kt
package org.delivery.db.userorder

import org.delivery.db.userorder.enums.UserOrderStatus
import org.springframework.data.jpa.repository.JpaRepository

interface UserOrderRepository : JpaRepository<UserOrderEntity, Long> {
    fun findAllByUserIdAndStatusOrderByIdDesc(userId: Long?, status: UserOrderStatus?): List<UserOrderEntity>

    fun findAllByUserIdAndStatusInOrderByIdDesc(
        userId: Long?,
        statusList: List<UserOrderStatus>
    ): List<UserOrderEntity>

    fun findByIdAndUserId(id: Long?, userId: Long?): UserOrderEntity?

    // 특정 주문
    // select * from user_order where id = ? and status = ? and user_id = ?
    fun findAllByIdAndStatusAndUserId(id: Long?, status: UserOrderStatus?, userId: Long?): UserOrderEntity
    fun findAllByIdAndUserId(id: Long?, userOd: Long?): UserOrderEntity?
}
